dbt/adapters/databricks/connections.py

```diff
@@ -47,15 +53,22 @@ def __pre_deserialize__(cls, data: Dict[Any, Any]) -> Dict[Any, Any]:
         return data
 
     def __post_init__(self) -> None:
-        # spark classifies database and schema as the same thing
-        if self.database is not None and self.database != self.schema:
+        session_properties = self.session_properties or {}
+        if CATALOG_KEY_IN_SESSION_PROPERTIES in session_properties:
+            if self.database is None:
+                self.database = session_properties[CATALOG_KEY_IN_SESSION_PROPERTIES]
+                del session_properties[CATALOG_KEY_IN_SESSION_PROPERTIES]
+            else:
+                raise dbt.exceptions.RuntimeException(
+                    f"`catalog` or `database` and `{CATALOG_KEY_IN_SESSION_PROPERTIES}` "
+                    "in `session_properties` cannot both be set"
+                )
+        self.session_properties = session_properties
+
+        if self.database is not None and not (self.database.strip()):
```

Comment:
Is this checking that the catalog name cannot be empty? How about changing the error message to be: `Invalid catalog name: {self.database}`?